src/iosMain/kotlin/com/airthings/lib/logging/platform/PlatformFileInputOutputImpl.kt

```diff
@@ -219,3 +216,12 @@ private fun <T> nsErrorWrapper(
         return if (error == null) result else fallback
     }
 }
+
+fun String.asNSData(): NSData = encodeToByteArray().asNSData()
+
+fun ByteArray.asNSData(): NSData = memScoped {
+    NSData.create(
+        bytes = allocArrayOf(this@asNSData),
+        length = this@asNSData.size.toULong(),
+    )
+}
```

Comment:
NIT: Not fond of using `this@label`, how about this?

```suggestion
fun ByteArray.asNSData(): NSData {
    val bytes = this
    return memScoped {
        NSData.create(
            bytes = allocArrayOf(bytes),
            length = bytes.size.toULong(),
        )
    }
}
```

Reply:
What's the problem with `this@label` here?

Reply:
Absolutely none :)

Reply:
@airxnoor  It was not meant to discredit the suggestion, I was merely wondering about your concern :)